### PR TITLE
[Polygon Edge] Permission smart contract deployment

### PR DESCRIPTION
**Pull request by Aleksao998**
**Description:**
This PR adds doc for permission smart contract deployment introduced with https://github.com/0xPolygon/polygon-edge/pull/692 PR

### CODE REVIEW

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  - smart contract
+  - permission
+  - deployment
+---
+
+## Overview
+
+This guide goes into detail on how to whitelist addresses that can deploy smart contracts. <br/> 
+Sometimes network operators wish to prevent users from deploying smart contracts that are unrelated to the network's purpose. Network operators can whitelist addresses that can deploy contracts in order to do so. Network operators can add or remove addresses. If the whitelist is empty any address can do the deployment, if not only addresses from the deployment whitelist can. 
+
```
Comment (laviniat1996) [inline docs/edge/additional-features/permission-contract-deployment.md:18]: I would suggest using some commas for the last phrase: "If the whitelist is empty, any address can do the deployment, if not, only addresses from the deployment whitelist can."

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+1. Execute the same command on all nodes
+2. Share modified `genesis.json`
+
+:::
+
+## How to use it?
+
+
+You can find all cli commands related to the deployment whitelist in the [CLI Commands](cli-commands#whitelist-commands) page.
+
```
Comment (laviniat1996) [inline docs/edge/additional-features/permission-contract-deployment.md:33]: The link is broken. This should work: [CLI Commands](/docs/edge/get-started/cli-commands#whitelist-commands) .

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Contract deployment whitelist : [0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d],
+
+
+```
+
+#### Add address to the deployment whitelist
+
+To add a new address to the deployment whitelist execute `whitelist deployment --add` CLI command. You can add as many addresses as you want. Only addresses that exists in the contract deployment whitelist can deploy contracts. If the whitelist is empty anyone can do the deployment.
+
```
Comment (laviniat1996) [inline docs/edge/additional-features/permission-contract-deployment.md:58]: I would suggest "Only the addresses that exist" and "If the whitelist is empty, anyone".

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+Contract deployment whitelist : [0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d 0x30ea4435167Ee91f9f874b5a894F3282A956C3FF],
+
+
+
+```
+
+#### Remove address from the deployment whitelist
+
+To remove an address from the deployment whitelist execute `whitelist deployment --remove` CLI command. Only addresses that exists in the contract deployment whitelist can deploy contracts. If the whitelist is empty anyone can do the deployment.
+
```
Comment (laviniat1996) [inline docs/edge/additional-features/permission-contract-deployment.md:76]: Here too I would suggest "Only the addresses that exist" and "If the whitelist is empty, anyone".

File: docs/edge/get-started/cli-commands.md [hunk 5]
```diff
@@ -1656,6 +1656,67 @@ message BlockchainEvent {
...
+  </TabItem>
+  <TabItem value="example" label="Example">
+
+    whitelist deployment --add 0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d
+
+  </TabItem>
+</Tabs>
+
+Adds a new address to the contract deployment whitelist. Only addresses in the contract deployment whitelist can deploy contracts, if empty anyone can.
+
```
Comment (laviniat1996) [inline docs/edge/get-started/cli-commands.md:1698]: Maybe it would be more clear to say "Only the addresses in the contract deployment whitelist can deploy contracts. If empty, anyone can."

File: docs/edge/get-started/cli-commands.md [hunk 6]
```diff
@@ -1656,6 +1656,67 @@ message BlockchainEvent {
...
+  </TabItem>
+  <TabItem value="example" label="Example">
+
+    whitelist deployment --remove 0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d
+
+  </TabItem>
+</Tabs>
+
+Removes a new address from the contract deployment whitelist. Only addresses in the contract deployment whitelist can deploy contracts, if empty anyone can.
+
```
Comment (laviniat1996) [inline docs/edge/get-started/cli-commands.md:1717]: Same here: "Only the addresses in the contract deployment whitelist can deploy contracts. If empty, anyone can."

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 1 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md:18]: Fixed 8b1739d5719304b72cc4b20b21ed979995bf6a05

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 2 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md:33]: Fixed 8b1739d5719304b72cc4b20b21ed979995bf6a05

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 3 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md:58]: Fixed 8b1739d5719304b72cc4b20b21ed979995bf6a05

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 4 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md:76]: Fixed 8b1739d5719304b72cc4b20b21ed979995bf6a05

File: docs/edge/get-started/cli-commands.md (hunk 5 above)
Comment (Aleksao998) [reply docs/edge/get-started/cli-commands.md:1698]: Fixed 8b1739d5719304b72cc4b20b21ed979995bf6a05

File: docs/edge/get-started/cli-commands.md (hunk 6 above)
Comment (Aleksao998) [reply docs/edge/get-started/cli-commands.md:1717]: Fixed 8b1739d5719304b72cc4b20b21ed979995bf6a05

**Review by laviniat1996 — APPROVED** [review]

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  - polygon
+  - edge
+  - smart contract
+  - permission
+  - deployment
+---
+
+## Overview
+
+This guide goes into detail on how to whitelist addresses that can deploy smart contracts. <br/>
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:16]: [This has got to be the longest paragraph I've read in my life that explains something so simple](https://www.youtube.com/watch?v=Ir0GoJyg8NA), it should've literally been:

This guide goes into detail on how to whitelist addresses that can deploy smart contracts.
Sometimes network operators wish to prevent users from deploying smart contracts that are unrelated to the network's purpose. Network operators can:
- Whitelist addresses for Smart Contract deployment
- Remove addresses from the whitelist for Smart Contract deployment

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+This guide goes into detail on how to whitelist addresses that can deploy smart contracts. <br/> 
+Sometimes network operators wish to prevent users from deploying smart contracts that are unrelated to the network's purpose. Network operators can whitelist addresses that can deploy contracts in order to do so. Network operators can add or remove addresses. If the whitelist is empty, any address can do the deployment, if not, only addresses from the deployment whitelist can.
+
+:::warning How to apply the changes
+
+This version of the permission contract deployment works off-chain. After updating deployment whitelists, nodes should align so that everyone has an updated whitelist. It is also necessary to restart the nodes in order for the changes to take effect. <br/>
+
+To share an updated whitelist you have 2 options:
+1. Execute the same command on all nodes
+2. Share modified `genesis.json`
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:25]: Typo, should be `Share the modified...`

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+This guide goes into detail on how to whitelist addresses that can deploy smart contracts. <br/> 
+Sometimes network operators wish to prevent users from deploying smart contracts that are unrelated to the network's purpose. Network operators can whitelist addresses that can deploy contracts in order to do so. Network operators can add or remove addresses. If the whitelist is empty, any address can do the deployment, if not, only addresses from the deployment whitelist can.
+
+:::warning How to apply the changes
+
+This version of the permission contract deployment works off-chain. After updating deployment whitelists, nodes should align so that everyone has an updated whitelist. It is also necessary to restart the nodes in order for the changes to take effect. <br/>
+
+To share an updated whitelist you have 2 options:
+1. Execute the same command on all nodes
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:24]: Why do you need to execute this on all nodes?
You only need to execute it on one node, and just share the genesis json that's been modified

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  - permission
+  - deployment
+---
+
+## Overview
+
+This guide goes into detail on how to whitelist addresses that can deploy smart contracts. <br/> 
+Sometimes network operators wish to prevent users from deploying smart contracts that are unrelated to the network's purpose. Network operators can whitelist addresses that can deploy contracts in order to do so. Network operators can add or remove addresses. If the whitelist is empty, any address can do the deployment, if not, only addresses from the deployment whitelist can.
+
+:::warning How to apply the changes
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:19]: Honestly, I would drop this warning section completely, as you explain in detail how this works below

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+2. Share modified `genesis.json`
+
+:::
+
+## How to use it?
+
+
+You can find all cli commands related to the deployment whitelist in the [CLI Commands](/docs/edge/get-started/cli-commands#whitelist-commands) page.
+
+* `whitelist show`: Displays whitelist information
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:34]: Hold on, why doesn't this command take in the path to the genesis json?

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+#### Show all addresses in the deployment whitelist
+
+There are 2 ways to find addresses from the deployment whitelist.
+1. Looking at the `genesis.json` where whitelists are saved
+2. Executing `whitelist show`, which prints information for all whitelists supported by Polygon Edge
+
+```bash
+
+polygon-edge % ./polygon-edge whitelist show
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:46]: What is this `%`?

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+[CONTRACT DEPLOYMENT WHITELIST]
+
+Added addresses: [0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d 0x30ea4435167Ee91f9f874b5a894F3282A956C3FF],
+Contract deployment whitelist : [0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d 0x30ea4435167Ee91f9f874b5a894F3282A956C3FF],
+
+
+
+```
+
+#### Remove address from the deployment whitelist
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:73]: Typo here, should be `Remove an address`

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+To remove an address from the deployment whitelist execute `whitelist deployment --remove` CLI command. Only addresses that exists in the contract deployment whitelist can deploy contracts. If the whitelist is empty, anyone can do the deployment.
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:75]: Typo, should be `only addresses that exist...`

Also, I would rephrase the last part `If the whitelist is empty, any address can do the deployment`

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+```bash
+
+polygon-edge % ./polygon-edge whitelist deployment --remove 0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d --remove 0x30ea4435167Ee91f9f874b5a894F3282A956C3FF
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:79]: Again, the weird %

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+[WHITELISTS]
+
+Contract deployment whitelist : [0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d],
+
+
+```
+
+#### Add address to the deployment whitelist
+
+To add a new address to the deployment whitelist execute `whitelist deployment --add` CLI command. You can add as many addresses as you want. Only addresses that exists in the contract deployment whitelist can deploy contracts. If the whitelist is empty, anyone can do the deployment.
```
Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:57]: Please specify that this flag expects an address as an argument

Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:57]: I would also rephrase this to be `There is no limit to the number of addresses present in the whitelist`, instead of `You can add as many...`

Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:57]: Typo here, should be `only addresses that exist in the...`

Comment (zivkovicmilos) [inline docs/edge/additional-features/permission-contract-deployment.md:57]: `anyone can do the deployment.` -> `any address can do the deployment`

File: docs/edge/get-started/cli-commands.md [hunk 17]
```diff
@@ -1656,6 +1656,67 @@ message BlockchainEvent {
...
+|------------------------|-------------------------------------------------------------------------------------|
+| whitelist show         | Displays whitelist information                     |
+| whitelist deployment   | Updatates smart contract deployment whitelist |
+
+<h3> whitelist show </h3>
+
+<Tabs>
+  <TabItem value="syntax" label="Syntax" default>
+
+    whitelist show
```
Comment (zivkovicmilos) [inline docs/edge/get-started/cli-commands.md:1671]: Missing path flag?

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 12 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md:46]: lol on my mac thats how it looks

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 11 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md:34]: relax just forgot to add it :D

File: docs/edge/get-started/cli-commands.md [hunk 18]
```diff
@@ -1656,6 +1656,67 @@ message BlockchainEvent {
...
+
+  </TabItem>
+  <TabItem value="example" label="Example">
+
+    whitelist deployment --add 0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d
+
+  </TabItem>
+</Tabs>
+
+Adds a new address to the contract deployment whitelist. Only the addresses in the contract deployment whitelist can deploy contracts. If empty, anyone can.
```
Comment (zivkovicmilos) [inline docs/edge/get-started/cli-commands.md:1697]: If empty, any address can execute the contract deployment

File: docs/edge/get-started/cli-commands.md [hunk 19]
```diff
@@ -1656,6 +1656,67 @@ message BlockchainEvent {
...
+
+  </TabItem>
+  <TabItem value="example" label="Example">
+
+    whitelist deployment --remove 0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d
+
+  </TabItem>
+</Tabs>
+
+Removes a new address from the contract deployment whitelist. Only the addresses in the contract deployment whitelist can deploy contracts. If empty, anyone can.
```
Comment (zivkovicmilos) [inline docs/edge/get-started/cli-commands.md:1716]: Removes an address, not a new address 🙂

Comment (zivkovicmilos) [inline docs/edge/get-started/cli-commands.md:1716]: If empty, any address can execute the contract deployment

**Review by zivkovicmilos — APPROVED** [review]
Lookin' good 💯

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 7 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Hahaha good one 7740695bf27c4a3dc9a094b8666b4646e34651d1

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 8 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed 929d3f75539f1aa3225cec4f640a72611c64d71a

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 9 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed 2f8b1fa0cda37640b60ce0c42f3afc53dbbe0e82

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 10 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed 2f8b1fa0cda37640b60ce0c42f3afc53dbbe0e82

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 12 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md:39]: isn't that a sign when you open a terminal to distinguish laptop name and command

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 15 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md:72]: check comment above

File: docs/edge/additional-features/permission-contract-deployment.md [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+polygon-edge % ./polygon-edge whitelist show 
+
+[WHITELISTS]
+
+Contract deployment whitelist : [0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d],
+
+
+```
+
+#### Add address to the deployment whitelist
```
[the remark this answers is not in the document]
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed 11f90eccc4afa0df9c7ec8fe0b8a761fb486734a

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 13 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed 11f90eccc4afa0df9c7ec8fe0b8a761fb486734a

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 14 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: fixed 0f5f9a76830d5e3a6edb3bacc29196a4b5646fa5

[the remark this answers is not in the document]
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed 3ccd429d78dfdbd07a0a14a8a3421c3528cde2fa

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 16 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed 3ccd429d78dfdbd07a0a14a8a3421c3528cde2fa

Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed 0f5f9a76830d5e3a6edb3bacc29196a4b5646fa5

Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed 7a1e1b1afcca36e78673ae9fdd290ac7d13465e0

Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: fixed 0f5f9a76830d5e3a6edb3bacc29196a4b5646fa5

File: docs/edge/get-started/cli-commands.md (hunk 19 above)
Comment (Aleksao998) [reply docs/edge/get-started/cli-commands.md]: Fixed 203078caef7ac4d80d3ea3bd14db6752405b96e0

Comment (Aleksao998) [reply docs/edge/get-started/cli-commands.md]: Fixed 203078caef7ac4d80d3ea3bd14db6752405b96e0

File: docs/edge/get-started/cli-commands.md (hunk 18 above)
Comment (Aleksao998) [reply docs/edge/get-started/cli-commands.md]: Fixed 203078caef7ac4d80d3ea3bd14db6752405b96e0

File: docs/edge/get-started/cli-commands.md [hunk 21]
```diff
@@ -1656,6 +1656,67 @@ message BlockchainEvent {
 
 ## Utilities
 
+### whitelist commands
+
+| **Command**            | **Description**                                                                     |
+|------------------------|-------------------------------------------------------------------------------------|
+| whitelist show         | Displays whitelist information                     |
+| whitelist deployment   | Updatates smart contract deployment whitelist |
```
[the remark this answers is not in the document]
Comment (Aleksao998) [reply docs/edge/get-started/cli-commands.md]: Fixed 203078caef7ac4d80d3ea3bd14db6752405b96e0

File: docs/edge/get-started/cli-commands.md (hunk 17 above)
Comment (Aleksao998) [reply docs/edge/get-started/cli-commands.md:1671]: Fixed abe3cb66a9aeb0af99026d1d7caa3784591510d3

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 11 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md:27]: fixed abe3cb66a9aeb0af99026d1d7caa3784591510d3

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 12 above)
Comment (zivkovicmilos) [reply docs/edge/additional-features/permission-contract-deployment.md:39]: I would just drop the % and everything before it

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 15 above)
Comment (zivkovicmilos) [reply docs/edge/additional-features/permission-contract-deployment.md:72]: check comment above 🙂

File: docs/edge/get-started/cli-commands.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (zivkovicmilos) [reply docs/edge/get-started/cli-commands.md]: You still have the typo `Updatates`

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 12 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed eb01734d7fd1f1304f1eff6f7bb3e11379461f16

File: docs/edge/additional-features/permission-contract-deployment.md (hunk 15 above)
Comment (Aleksao998) [reply docs/edge/additional-features/permission-contract-deployment.md]: Fixed, check comment above 😃

File: docs/edge/get-started/cli-commands.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (Aleksao998) [reply docs/edge/get-started/cli-commands.md]: my mistake b401532c2c1a913be23287c68974fc131f766f8e
